app/code/Magento/Catalog/Pricing/Price/TierPrice.php

```diff
@@ -157,6 +163,9 @@ function (&$priceData) {
                     /* convert string value to float */
                     $priceData['price_qty'] = $priceData['price_qty'] * 1;
                     $priceData['price'] = $this->applyAdjustment($priceData['price']);
+                    if ($this->scopeConfig->getValue('tax/display/type') == 3) {
```

Comment:
Please, use const `\Magento\Tax\Model\Config::DISPLAY_TYPE_BOTH` instead of int, also use strict comparison here.